compose/foundation/foundation/src/uikitMain/kotlin/androidx/compose/foundation/cupertino/CupertinoOverscrollEffect.kt

```diff
@@ -60,14 +61,20 @@ private data class CupertinoOverscrollAvailableDelta(
     val newOverscrollValue: Float
 )
 
+/**
+ * CupertinoOverscrollEffect
+ *
+ * @param density to be taken into consideration during computations;
+ * Cupertino formulas use DPs, and scroll machinery uses raw values.
+ *
+ * @param applyClip Some consumers of overscroll effect apply clip by themselves and some don't,
+ * thus this flag is needed to update our modifier chain and make the clipping correct in every case while avoiding redundancy
+ */
 @OptIn(ExperimentalFoundationApi::class)
 class CupertinoOverscrollEffect(
-    /*
-     * Density to be taken into consideration during computations; Cupertino formulas use
-     * DPs, and scroll machinery uses raw values.
-     */
     private val density: Float,
-    layoutDirection: LayoutDirection
+    layoutDirection: LayoutDirection,
+    val applyClip: Boolean
```

Comment:
1. Can we pass base modifier instead of bool flag to be more flexible here? Just a topic for discussion, not sure what's the best option here
2. Should we do expose it in public API at all? Can we do it with composition locals like Android? See `LocalOverscrollConfiguration`

Reply:
Hm, I'm not sure we should pass Modifier here, because that's too intrusive.
The second point is still discussed with Google, `ScrollConfiguration` can become common API.